app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
`.select`? Don't you mean `.each`? Also, This will delete EVERY server on the account.  This would include any non-proxy servers that you may or may not have.  This should be part of fog.rb.  Something along the lines of
```ruby
connection.servers.each do |s|
  if server_is_proxy_type? s
    s.destroy
  end
  proxies.active.each do |p|
    self.soft_delete
    self.save
  end
end
```
This way every fog decedent gets this method, not just joyent.